cpp/models/ode_seair/parameters.h

```diff
@@ -0,0 +1,217 @@
+struct SocialDistancing {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.2);
+    }
+    static std::string name()
+    {
+        return "SocialDistancing";
+    }
+};
+
+/**
+ * @brief Quarantining.
+ */
+template <typename FP = double>
+struct Quarantined {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.2);
+    }
+    static std::string name()
+    {
+        return "Quarantined";
+    }
+};
+
+/**
+ * @brief Rate of testing.
+ */
+template <typename FP = double>
+struct TestingRate {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.2);
+    }
+    static std::string name()
+    {
+        return "TestingRate";
+    }
+};
+
+/**
+ * @brief Recovery rate.
+ */
+template <typename FP = double>
+struct RecoveryRate {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.0067);
+    }
+    static std::string name()
+    {
+        return "RecoveryRate";
+    }
+};
+
+/**
+ * @brief Death Rate.
+ */
+template <typename FP = double>
+struct DeathRate {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.0041);
+    }
+    static std::string name()
+    {
+        return "DeathRate";
+    }
+};
+
+/**
+ * @brief Inverse of the latent period of the virus.
+ */
+template <typename FP = double>
+struct TimeExposed {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.5);
+    }
+    static std::string name()
+    {
+        return "TimeExposed";
+    }
+};
+
+/**
+ * @brief Infectious period for unconfirmed infected people.
+ */
+template <typename FP = double>
+struct RecoveryRateFromAsymptomatic {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.1);
+    }
+    static std::string name()
+    {
+        return "RecoveryRateFromAsymptomatic";
+    }
+};
+
+/**
+ * @brief Rate recovered people become susceptible again.
+ */
+template <typename FP = double>
+struct TimeRecoveredInv {
+    using Type = FP;
+    static Type get_default()
+    {
+        return Type(0.0);
+    }
+    static std::string name()
+    {
+        return "TimeRecoveredInv";
+    }
+};
+
+template <typename FP = double>
+using ParametersBase =
+    ParameterSet<SocialDistancing<FP>, Quarantined<FP>, TestingRate<FP>, RecoveryRate<FP>, DeathRate<FP>,
+                 TimeExposed<FP>, RecoveryRateFromAsymptomatic<FP>, TimeRecoveredInv<FP>>;
+
+/**
+ * @brief Parameters of an SEAIR model.
+ */
+template <typename FP = double>
+class Parameters : public ParametersBase<FP>
+{
+public:
+    Parameters()
+        : ParametersBase<FP>()
+    {
+    }
+
+    /**
+     * @brief Checks whether all Parameters satisfy their corresponding constraints and logs an error 
+     * if constraints are not satisfied.
+     * @return Returns 1 if one constraint is not satisfied, otherwise 0.   
+     */
+    int check_constraints() const
+    {
+        return 0;
```

Comment:
Should we add constraints here before merging?

Reply:
copy/pasting and reducing from secir should be realistic in 10minutes, am I wrong?